src/tensorwaves/optimizer/minuit.py

```diff
@@ -16,27 +16,23 @@ class Minuit2(Optimizer):
     Implements the `~.interfaces.Optimizer` interface.
     """
 
-    def __init__(self) -> None:
-        pass
+    def __init__(self, callback: Optional[Callback] = None) -> None:
+        self.__callback = callback
 
     def optimize(self, estimator: Estimator, initial_parameters: dict) -> dict:
         parameters = initial_parameters
-        progress_bar = tqdm()
 
-        @tf_file_logging(iterations=2)
-        def __call_estimator(params: dict) -> float:
-            estimator.update_parameters(params)
-            progress_bar.update()
-            return estimator()
-
-        def __func(pars: list) -> float:
-            """Wrap the estimator."""
+        def __wrapped_function(pars: list) -> float:
             for i, k in enumerate(parameters.keys()):
                 parameters[k] = pars[i]
-            return __call_estimator(parameters)
+            estimator.update_parameters(parameters)
+            estimator_value = estimator()
+            if self.__callback is not None:
```

Comment:
I just realized that this if case at that point is a bit smelly. I'm not sure how smart python is in optimizing this. But we could improve this by checking at the init level if callbacks are given and if so we call a second version of a `__wrapped_function` which includes the callback.

Reply:
Alternative: set an empty `CallbackList` in case of `None` in `Minuit2.__init__`

Reply:
7f02297